IQSS/dataverse-frontend
# 356 publish dataset

**What this PR does / why we need it**:

**Which issue(s) this PR closes**:

Closes #356 

**Special notes for your reviewer**:
I wasn't able to add the major and minor version numbers to the popup, because they aren't currently part of the Dataset model.  I added an issue for this: https://github.com/IQSS/dataverse-frontend/issues/428

**Suggestions on how to test this**:
Review the PublishDatasetModal Stories: http://localhost:6006/?path=/story/sections-dataset-page-publishdatasetmodal--default
Steps to test in the SPA:

1. Create a new dataset, which will be a Draft version.
2. Go to the Dataset Page, Click Publish Dataset Menu, and Publish dropdown
3. Click continue
4. The Publish Dataset should close and the "Publish in Progress" alert will appear on the Dataset Page.
5. When the publish is completed, the "Publish in Progress" will disappear and the Dataset will be version 1.0
6. Edit the Dataset Metadata and choose publish again.
7. Now the Publish Dataset Model will give an option of Major or Minor release.  
8. Choose the version update type, and click continue.
9. The Dataset will be published again with the new version chosen (either + 1.0 for Major update, or + .1 for Minor update

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

**Is there a release notes update needed for this change?**:

**Additional documentation**:
